packages/task/src/browser/quick-open-task.ts

```diff
@@ -56,20 +56,16 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
 
     /** Initialize this quick open model with the tasks. */
     async init(): Promise<void> {
+        const recentTasks = await this.taskService.getRecentTasks();
         const configuredTasks = this.taskService.getConfiguredTasks();
         const providedTasks = await this.taskService.getProvidedTasks();
 
-        const filteredProvidedTasks: TaskConfiguration[] = [];
-        providedTasks.forEach(provided => {
-            if (!configuredTasks.some(configured => configured.label === provided.label)) {
-                filteredProvidedTasks.push(provided);
-            }
-        });
-
+        const { filteredRecentTasks, filteredConfiguredTasks, filteredProvidedTasks } = this.getFilteredTasks(recentTasks, configuredTasks, providedTasks);
         this.items = [];
         this.items.push(
-            ...configuredTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, true, ind === 0 ? 'configured tasks' : undefined)),
-            ...filteredProvidedTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, false, ind === 0 ? 'detected tasks' : undefined))
+            ...filteredRecentTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, ind === 0 ? 'recently used tasks' : undefined)),
+            ...filteredConfiguredTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, ind === 0 ? 'configured tasks' : undefined)),
```

Comment:
'`showBorder`:  true' will be nice, but you need to test different combinations, i.e. configured tasks should not have it if there are no recently used tasks and detected tasks should not have it if there are not recent and configured tasks

btw please use option pattern to avoid adding and removing new args, like here for example: 
https://github.com/elaihau/theia/blob/ce2f145017ff6bca715cddb96f876366eb1a4b22/packages/file-search/src/browser/quick-file-open.ts#L336-L340

Reply:
Vincent is currently working on the border part: https://github.com/theia-ide/theia/issues/4770. I will leave it in his capable hands :)